app/controllers/declaration/LinkDucrToMucrController.scala

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2021 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers.declaration
+
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+
+import controllers.actions.{AuthAction, JourneyAction}
+import controllers.navigation.Navigator
+import forms.common.YesNoAnswer
+import forms.common.YesNoAnswer.{form, YesNoAnswers}
+import models.requests.JourneyRequest
+import models.{DeclarationType, ExportsDeclaration, Mode}
+import play.api.i18n.I18nSupport
+import play.api.mvc.{Action, AnyContent, Call, MessagesControllerComponents}
+import services.cache.ExportsCacheService
+import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendController
+import views.html.declaration.link_ducr_to_mucr
+
+@Singleton
+class LinkDucrToMucrController @Inject()(
+  authenticate: AuthAction,
+  journeyType: JourneyAction,
+  override val exportsCacheService: ExportsCacheService,
+  navigator: Navigator,
+  mcc: MessagesControllerComponents,
+  linkDucrToMucrPage: link_ducr_to_mucr
+)(implicit ec: ExecutionContext)
+    extends FrontendController(mcc) with I18nSupport with ModelCacheable with SubmissionErrors {
+
+  private val yesNoForm = form()
```

Comment:
Not sure is this is correct, as the same instance is shared between all calls. Probably the only reason we are getting away with this is because both `withSubmissionErrors()` and `bindFromRequest()` return a new instance.
Was very confusing to me and I had to check manually if it poses any danger.

Reply:
:+1: